### PR TITLE
Implement Api client and Api calls 

### PR DESCRIPTION
**Pull request by kubaolszewski**

### CODE REVIEW

File: lib/app/custom_widgets/live_match_tile.dart [hunk 1]
```diff
@@ -1,16 +1,22 @@
...
   const LiveMatchTile({
     super.key,
+    required this.liveMatch,
   });
 
+  final LiveMatchModel liveMatch;
+
   @override
   Widget build(BuildContext context) {
+    final String assetName = liveMatch.league!.flag!;
```
Comment (MBanaszynski) [inline lib/app/custom_widgets/live_match_tile.dart]: Are you sure it will never be null?

File: lib/app/custom_widgets/live_match_tile.dart [hunk 2]
```diff
@@ -23,93 +29,155 @@ class LiveMatchTile extends StatelessWidget {
...
+                              ),
+                            ),
+                          ),
                         ),
+                        const SizedBox(width: 10),
                         Text(
-                          '78',
-                          style: TextStyle(color: Colors.green),
-                        ),
+                          liveMatch.league!.name!,
```
Comment (MBanaszynski) [inline lib/app/custom_widgets/live_match_tile.dart]: Same here

File: lib/app/custom_widgets/live_match_tile.dart [hunk 3]
```diff
@@ -23,93 +29,155 @@ class LiveMatchTile extends StatelessWidget {
...
+                      ),
+                      child: Row(
+                        mainAxisAlignment: MainAxisAlignment.spaceAround,
+                        children: [
+                          const CircleAvatar(
+                            radius: 5,
+                            backgroundColor: Colors.green,
+                          ),
+                          Text(
+                            liveMatch.fixture!.status!.short!,
```
Comment (MBanaszynski) [inline lib/app/custom_widgets/live_match_tile.dart]: Same here

File: lib/app/custom_widgets/live_match_tile.dart [hunk 4]
```diff
@@ -23,93 +29,155 @@ class LiveMatchTile extends StatelessWidget {
...
+                      ),
+                      Expanded(
+                        flex: 1,
+                        child: SizedBox.square(
+                          child: Column(
+                            mainAxisAlignment: MainAxisAlignment.center,
+                            children: [
+                              Image(
+                                image: NetworkImage(
+                                  liveMatch.teams!.away!.logo!,
```
Comment (MBanaszynski) [inline lib/app/custom_widgets/live_match_tile.dart]: Same here

File: lib/app/custom_widgets/live_match_tile.dart [hunk 5]
```diff
@@ -23,93 +29,155 @@ class LiveMatchTile extends StatelessWidget {
...
-                  Container(
-                    height: 30,
-                    width: 50,
-                    decoration: BoxDecoration(
-                      borderRadius: BorderRadius.circular(24),
-                      color: AppColors.liveTimerBackground,
-                    ),
-                    child: const Row(
-                      mainAxisAlignment: MainAxisAlignment.spaceAround,
+              Expanded(
```
Comment (MBanaszynski) [inline lib/app/custom_widgets/live_match_tile.dart]: Please move all "sections" to separated files, e.g. `live_match_tile_title`, `live_match_tile_logo` etc.

File: lib/app/custom_widgets/wide_match_tile.dart [hunk 6]
```diff
@@ -1,110 +1,137 @@
...
-                        ),
-                      )
-                    ],
-                  ),
-                  SizedBox(height: 10),
-                  Row(
-                    children: [
-                      CircleAvatar(
-                        radius: 15,
+                        liveMatch.fixture!.status!.short!,
```
Comment (MBanaszynski) [inline lib/app/custom_widgets/wide_match_tile.dart]: Same here

File: lib/app/custom_widgets/wide_match_tile.dart [hunk 7]
```diff
@@ -1,110 +1,137 @@
...
+                          )
+                        ],
                       ),
+                      const SizedBox(height: 10),
+                      Row(
+                        children: [
+                          Image(
+                            width: 35,
+                            height: 35,
+                            image: NetworkImage(liveMatch.teams!.away!.logo!),
```
Comment (MBanaszynski) [inline lib/app/custom_widgets/wide_match_tile.dart]: Same here

File: lib/app/custom_widgets/wide_match_tile.dart [hunk 8]
```diff
@@ -1,110 +1,137 @@
...
-                        style: TextStyle(
-                          color: Colors.grey,
-                          fontSize: 16,
-                          fontWeight: FontWeight.bold,
-                        ),
+                      Row(
+                        mainAxisAlignment: MainAxisAlignment.end,
+                        children: [
+                          Text(
+                            '${liveMatch.goals!.home}',
```
Comment (MBanaszynski) [inline lib/app/custom_widgets/wide_match_tile.dart]: Why did you use '$...'?

File: lib/app/features/home/home_widgets/live_now_view/live_now_view.dart [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+            child: CircularProgressIndicator(
+              color: AppColors.mainThemePink,
+            ),
+          );
+        }
+
+        final liveMatches = state.liveMatchModel;
+
+        if (liveMatches!.isEmpty) {
+          return const SizedBox(
```
Comment (MBanaszynski) [inline lib/app/features/home/home_widgets/live_now_view/live_now_view.dart]: Consider moving to separated file as `empty_list_placeholder.dart`

File: lib/app/features/home/home_widgets/live_now_view/live_now_view_module.dart [hunk 10]
```diff
@@ -0,0 +1,3 @@
+import 'package:flutter_modular/flutter_modular.dart';
+
```
Comment (MBanaszynski) [inline lib/app/features/home/home_widgets/live_now_view/live_now_view_module.dart:2]: Do you need this file?

File: lib/app/features/home/match_details/match_details.dart [hunk 11]
```diff
@@ -1,32 +1,42 @@
...
 
-  final _detailsTitle = const DetailsTitle();
+  final _detailsTitle = const _DetailsTitle();
+  final LiveMatchModel liveMatch;
 
   @override
   Widget build(BuildContext context) {
-    return BlocProvider(
-      create: (context) => HomeCubit(),
+    final homeCubit = Modular.get<HomeCubit>();
```
Comment (MBanaszynski) [inline lib/app/features/home/match_details/match_details.dart]: Don't use `Modular.get<_cubit_>`

File: lib/app/features/home/match_details/match_details.dart [hunk 12]
```diff
@@ -52,19 +62,33 @@ class MatchDetails extends StatelessWidget {
...
+                                      ),
+                                    ),
+                                  ),
                                 ),
-                                SizedBox(width: 10),
+                                const SizedBox(width: 10),
                                 Text(
-                                  'League name',
-                                  style: TextStyle(color: Colors.grey, fontSize: 16, fontWeight: FontWeight.bold),
+                                  liveMatch.league!.name!,
```
Comment (MBanaszynski) [inline lib/app/features/home/match_details/match_details.dart]: Same here

File: lib/app/features/home/match_details/match_details.dart [hunk 13]
```diff
@@ -89,45 +113,67 @@ class MatchDetails extends StatelessWidget {
...
                             ),
-                            SizedBox(width: 30),
-                            Placeholder(fallbackHeight: 50, fallbackWidth: 50),
+                            const SizedBox(width: 30),
+                            Column(
+                              mainAxisAlignment: MainAxisAlignment.center,
+                              children: [
+                                Image(
+                                  image: NetworkImage(
+                                    liveMatch.teams!.away!.logo!,
```
Comment (MBanaszynski) [inline lib/app/features/home/match_details/match_details.dart]: Same here

**Review by MBanaszynski — CHANGES_REQUESTED** [review]

File: lib/app/features/home/home_widgets/live_now_view/live_now_view_module.dart (hunk 10 above)
Comment (kubaolszewski) [reply lib/app/features/home/home_widgets/live_now_view/live_now_view_module.dart:2]: Deleted it in progress

File: lib/app/features/home/home_widgets/live_now_view/live_now_view.dart (hunk 9 above)
Comment (kubaolszewski) [reply lib/app/features/home/home_widgets/live_now_view/live_now_view.dart]: done

File: lib/app/custom_widgets/live_match_tile.dart (hunk 5 above)
Comment (kubaolszewski) [reply lib/app/custom_widgets/live_match_tile.dart]: done

**Review by MBanaszynski — APPROVED** [review]

**Pull request merged**